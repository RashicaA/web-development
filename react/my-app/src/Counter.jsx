// Task Overview:
// Build a simple Counter component where:
//     Buttons are provided to increment, decrement, and reset the count.
//     The initial count is passed as a prop to the Counter component.
//     The count cannot be decremented below 0.

import React, { useState } from 'react';

const Counter = ({ initialCount = 0 }) => {
    const [counter, setCounter] = useState(initialCount);

    function handleIncrement() {
        setCounter(counter + 1);
    }

    function handleDecrement() {
        if (counter > 0) {
            setCounter(counter - 1);
        }
    }

    function handleReset() {
        setCounter(initialCount);
    }

    return (
        <>
            <h4>Count: {counter}</h4>
            <button data-testid="increment" onClick={handleIncrement}>Increment</button>
            <button data-testid="decrement" onClick={handleDecrement}>Decrement</button>
            <button data-testid="reset" onClick={handleReset}>Reset</button>
        </>
    );
};


export default Counter;