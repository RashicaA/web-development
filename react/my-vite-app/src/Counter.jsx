// Task Overview:
// Build a simple Counter component where:
//     Buttons are provided to increment, decrement, and reset the count.
//     The initial count is passed as a prop to the Counter component.
//     The count cannot be decremented below 0.

import React, { useState } from 'react';

const Counter = () => {
    let [counter, setCounter] = useState(100);
    let count = 0;
    function handleIncrement() {
        // count++;
        // console.log(count)

        setCounter(counter + 1)

    }
    function handleDecrement() {
        // count--;
        setCounter(counter - 1)
    }
    function handleReset() {
        // count = 0;
        setCounter(0)
    }
    return <>
        <h4>Count: {counter}</h4>
        <button onClick={handleIncrement}>Increment</button>
        <button onClick={handleDecrement}>Decrement</button>
        <button onClick={handleReset}>Reset</button>
    </>
}

export default Counter;