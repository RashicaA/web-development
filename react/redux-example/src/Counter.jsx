import React from 'react'
import { increment, decrement } from './redux/counterSlice'
import { useDispatch, useSelector } from 'react-redux'

const Counter = () => {
    const dispatch = useDispatch();
    const count = useSelector((state) => state.counter.value)
    console.log("Counter component re-rendered..")
    return (
        <div>
            <button onClick={() => dispatch(decrement())}> - </button>
            <h3>{count}</h3>
            <button onClick={() => dispatch(increment())}> + </button>
        </div>
    )
}

export default Counter
