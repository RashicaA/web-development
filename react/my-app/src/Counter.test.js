import { render, screen, fireEvent } from "@testing-library/react";
import Counter from "./Counter";
import '@testing-library/jest-dom';

describe('Counter button tests', () => {
    test('Initial state check of buttons and count', () => {
        render(<Counter initialCount={0} />);
        expect(screen.getByText("Count: 0")).toBeInTheDocument();
        expect(screen.getByTestId("increment")).toBeInTheDocument();
        expect(screen.getByTestId("decrement")).toBeInTheDocument();
        expect(screen.getByTestId("reset")).toBeInTheDocument();
    });
    test("on increment btn click, count should increase by 1", () => {
        render(<Counter initialCount={0} />)
        const incrBtn = screen.getByTestId("increment");
        fireEvent.click(incrBtn);
        expect(screen.queryByText("Count: 1")).toBeInTheDocument();
        fireEvent.click(incrBtn);
        fireEvent.click(incrBtn);
        expect(screen.queryByText("Count: 3")).toBeInTheDocument();
    });
    test("on decrement btn click, count should decrease by 1", () => {
        render(<Counter initialCount={0} />)
        const incrBtn = screen.getByTestId("increment");
        const decrBtn = screen.getByTestId("decrement");
        fireEvent.click(incrBtn); // 1
        fireEvent.click(incrBtn); // 2
        fireEvent.click(incrBtn); //3
        fireEvent.click(decrBtn); //2
        expect(screen.queryByText("Count: 2")).toBeInTheDocument();
        fireEvent.click(decrBtn); //1
        expect(screen.queryByText("Count: 1")).toBeInTheDocument();
        fireEvent.click(decrBtn); //0
        expect(screen.queryByText("Count: 0")).toBeInTheDocument();
        fireEvent.click(decrBtn); // still remains as 0
        expect(screen.queryByText("Count: 0")).toBeInTheDocument();
    });

    test("on reset btn click, count should become 0", () => {
        render(<Counter initialCount={0} />)
        const incrBtn = screen.getByTestId("increment");
        const reset = screen.getByTestId("reset");
        fireEvent.click(incrBtn);
        fireEvent.click(incrBtn);
        fireEvent.click(incrBtn);
        fireEvent.click(reset);
        expect(screen.queryByText("Count: 0")).toBeInTheDocument();
    })

    test("snapshot for counter", () => {
        const { asFragment } = render(<Counter />);
        expect(asFragment()).toMatchSnapshot()
    })

});
