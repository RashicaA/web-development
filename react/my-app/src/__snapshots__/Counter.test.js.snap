// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter button tests snapshot for counter 1`] = `
<DocumentFragment>
  <h4>
    Count: 0
  </h4>
  <button
    data-testid="increment"
  >
    Increment
  </button>
  <button
    data-testid="decrement"
  >
    Decrement
  </button>
  <button
    data-testid="reset"
  >
    Reset
  </button>
</DocumentFragment>
`;
